addons/crm/models/crm_lead.py

```diff
@@ -1258,6 +1258,11 @@ def message_new(self, msg_dict, custom_values=None):
             through message_process.
             This override updates the document according to the email.
         """
+
+        # remove external users
+        if not self.env.user.has_group('crm.lead'):
+            self = self.with_context(default_user_id=self._context.get('uid'))
```

Comment:
What is the value of `self._context.get('uid')`? If I understand correctly, we are usually in a situation where the method is called through a cron. Therefore, I don't think the `uid` is a good option. I'd rather either pop the context key, or set to False as done below.

Reply:
ah indeed I was testing by manually calling the action, but in the context of an automatic execution there would be no uid I suppose. Setting the default_user_id is working, the lead will have no salesman assigned